pkg/zcapld/resolve.go

```diff
@@ -51,34 +53,48 @@ func (s SimpleKeyResolver) Resolve(keyID string) (*verifier.PublicKey, error) {
 	return key, nil
 }
 
-// DIDKeyResolver resolves verification keys from did:key URLs: https://w3c-ccg.github.io/did-method-key/.
-type DIDKeyResolver struct {
+type kmsProvider struct{ kms.KeyManager }
+
+func (p *kmsProvider) KMS() kms.KeyManager {
+	return p
+}
+
+// NewDIDResolver creates new DID resolver.
+func NewDIDResolver(k kms.KeyManager, v VDRResolver) *DIDResolver {
+	if v != nil {
+		return &DIDResolver{KMS: k, VDR: v}
+	}
+
+	return &DIDResolver{KMS: k, VDR: vdr.New(&kmsProvider{KeyManager: k}, vdr.WithVDR(vdrkey.New()))}
+}
+
+// DIDResolver resolves verification keys from did:key URLs: https://w3c-ccg.github.io/did-method-key/.
+type DIDResolver struct {
+	KMS kms.KeyManager
+	VDR VDRResolver
 }
 
 // Resolve expects 'didKeyURL' to be a did:key URL.
 // Example: "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH#z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH".
-func (d *DIDKeyResolver) Resolve(didKeyURL string) (*verifier.PublicKey, error) {
-	const numParts = 2
+func (d *DIDResolver) Resolve(id string) (*verifier.PublicKey, error) {
+	didToResolve := id
 
-	parts := strings.Split(didKeyURL, "#")
-	if len(parts) != numParts {
-		return nil, fmt.Errorf("not a did:key URL: %s", didKeyURL)
+	if strings.HasPrefix(id, "did:key") {
+		didToResolve = strings.Split(id, "#")[0]
```

Comment:
I believe the split needs to be done always because `id` is meant to be a DID URL with a fragment.